packages/fxa-graphql-api/src/lib/datasources/profileServer.ts

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { DataSource, DataSourceConfig } from 'apollo-datasource';
+import { Client } from 'fxa-js-client';
+import superagent from 'superagent';
+import { Container } from 'typedi';
+
+import { configContainerToken, fxAccountClientToken } from '../constants';
+import { Context } from '../server';
+
+export class ProfileServerSource extends DataSource {
+  private authClient: Client;
+  private oauthToken: String | undefined;
+  private oauthClientId: string;
+  private profileServerUrl: string;
+
+  constructor(private token: string = '') {
```

Comment:
Maybe more a note to myself, since I was confused that this `token` parameter didn't seem used anywhere: TIL [TypeScript supports parameter properties](https://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties), which take care of declaring & initializing an object property all at once.